### PR TITLE
Revamps docs for 5.0 #1248

### PR DESCRIPTION
**Pull request by slemeur**
Pull request to revamp the docs for 5.0

This PR:
- adds docs content to the repository as markdown files
- provides utilities Jekyll to build and run the docs to edit them locally
- provides styles for the static HTML site built with Jekyll
- packages and deploy the docs as a war file in the assembly
- integrates the docs inside of UD and IDE

See issue: #1248

### CODE REVIEW

**Comment by benoitf** [comment]
many file permissions have been changed. It shouldn't

File: docs/.gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+_site/
+*.swp
+pkg/
+test/
+.jekyll-metadata
+.sass-cache/
```
Comment (benoitf) [inline docs/.gitignore]: missing end of line

File: docs/_data/jekyllconf-talks.yml [hunk 2]
```diff
@@ -0,0 +1,167 @@
+- speaker: Ben Balter
```
Comment (benoitf) [inline docs/_data/jekyllconf-talks.yml]: are this content accurate ? all this speakers ??

File: docs/.gitignore (hunk 1 above)
Comment (slemeur) [inline docs/.gitignore]: fixed

**Comment by slemeur** [comment]
- Fixed the permissions for script files.
- Fixed the missing end of line in .gitignore
- Remove useless content

File: docs/.gitignore (hunk 1 above)
Comment (slemeur) [inline docs/.gitignore]: fixed

**Pull request merged**